### PR TITLE
Link tutorial page backend

### PR DESCRIPTION
**Pull request by s-hane**

### CODE REVIEW

File: backend/src/api/v1/tutorials/endpoints/putBook.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { getTutorial } from '../utils';
 
 const docClient = new AWS.DynamoDB.DocumentClient();
 
+const MSINDAY = 86400000;
```
Comment (ericgmoon) [inline backend/src/api/v1/tutorials/endpoints/putBook.ts:10]: MS_IN_DAY snake case for constants please

File: src/pages/TutorialsPage/index.tsx [hunk 2]
```diff
@@ -5,16 +5,14 @@ import WeekCalendar from '../../components/WeekCalendar';
 import { useGetTutorialsInRangeQuery } from '../../services/tutorials';
 
 const MSINWEEK = 604800000;
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:7]: camel case please

**Review by ericgmoon — APPROVED** [review]

File: src/services/tutorials/utils.ts [hunk 3]
```diff
@@ -27,15 +28,18 @@ const getStartDatetime = (startDatetimeIdentifier: string) =>
...
     return 'readOnly';
   }
-  if (getStartDatetime(tutorial.startDatetimeIdentifier) - Date.now() < 5 * MSINMIN) {
+  if (getStartDatetime(tutorial.startDatetimeIdentifier) - Date.now() < 5 * MS_IN_MIN) {
     return 'imminent';
   }
   if (tutorial.attendees.length >= tutorial.capacity) {
     return 'full';
   }
+  if (tutorial.attendees.some((student) => student.studentId === getCurrentUserId())) {
```
Comment (ericgmoon) [inline src/services/tutorials/utils.ts:40]: Would be good to write some tests for this util function

File: src/pages/TutorialsPage/index.tsx [hunk 4]
```diff
@@ -1,30 +1,114 @@
...
         <WeekCalendar startDatetime={new Date().getTime()} />
       </TitleWrapper>
       <TitleWrapper title="All Tutorials">
-        <WeekCalendar
-          startDatetime={new Date().getTime()}
-          data={physTutorials?.concat(chemTutorials ?? [])}
-        />
+        {
+          isLoading ? (
+            <LoadingWrapper>
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:102]: Use the `LoadingWrapper` component from `components/PageLayout`

Comment (ericgmoon) [reply src/pages/TutorialsPage/index.tsx:102]: For an example, see how I handle loading in `VideoPlayerPage`

File: src/components/WeekCalendar/types.ts [hunk 5]
```diff
@@ -5,6 +5,9 @@ export type CalendarEntry = {
   color?: 'primary' | 'chemistry' | 'physics',
   properties?: {[key: string]: string},
   state?: 'imminent' | 'open' | 'full' | 'booked' | 'readOnly',
+  onJoin?: () => {},
```
Comment (ericgmoon) [inline src/components/WeekCalendar/types.ts:8]: The issue with this approach is if `WeekCalendar` contains events of DIFFERENT types requiring different `onBook` functions, no?

File: src/pages/TutorialsPage/index.tsx [hunk 6]
```diff
@@ -1,30 +1,114 @@
...
+    startDatetime: Date.now() - 2 * MS_IN_HOUR, endDatetime: Date.now() + 2 * MS_IN_WEEK, courses: 'phys11,chem11',
   });
-  const [chemArgs] = useState({
-    startDatetime: Date.now() - 2 * 604800000, endDatetime: Date.now() + 2 * 604800000, course: 'chem11',
+  const createSnackbar = useSnackbar();
+
+  const getSubject = (courseCode: string) => {
+    switch (courseCode) {
+      case 'chem11':
+        return 'chemistry';
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:30]: you can delete this line

File: src/pages/TutorialsPage/index.tsx [hunk 7]
```diff
@@ -1,30 +1,114 @@
...
+      case 'chem12':
+        return 'chemistry';
+      case 'phys11':
+        return 'physics';
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:34]: you can delete this line too,

```
case 'phys11':
case'phys12':
return 'physics'
```

File: src/pages/TutorialsPage/index.tsx [hunk 8]
```diff
@@ -1,30 +1,114 @@
...
+      case 'phys12':
+        return 'physics';
+      default:
+        return undefined;
+    }
+  };
+
+  const toSentenceCase = (sentence: string) =>
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:42]: util function?

File: src/pages/TutorialsPage/index.tsx [hunk 9]
```diff
@@ -1,30 +1,114 @@
...
-const MSINWEEK = 604800000;
+const MS_IN_WEEK = 604800000;
+const MS_IN_DAY = 86400000;
+const MS_IN_HOUR = 3600000;
+const MS_IN_MIN = 60000;
 
 const TutorialsPage = () => {
-  const [physArgs] = useState({
-    startDatetime: Date.now() - 2 * MSINWEEK, endDatetime: Date.now() + 2 * MSINWEEK, course: 'phys11',
+  const [args] = useState({
```
Comment (ericgmoon) [inline src/pages/TutorialsPage/index.tsx:22]: If this is a static value, why do we need to use `useState`?

**Review by ericgmoon — CHANGES_REQUESTED** [review]

File: src/services/tutorials/utils.ts (hunk 3 above)
Comment (s-hane) [reply src/services/tutorials/utils.ts:40]: fine

File: src/services/tutorials/index.ts [hunk 10]
```diff
@@ -1,28 +1,47 @@
...
+        const { course, id, studentId } = arg;
+        return {
+          url: `tutorials/${course}/${id}/book`,
+          body: {
+            studentId,
+          },
+          method: 'PUT',
+        };
+      },
+      invalidatesTags: (arg: any) => [{ type: 'Tutorials', id: arg }],
```
Comment (ericgmoon) [inline src/services/tutorials/index.ts:42]: Shouldn't the id here be `arg.id`?

File: src/services/tutorials/index.ts [hunk 11]
```diff
@@ -1,28 +1,47 @@
...
 export const tutorialsApi = cognitoApi.injectEndpoints({
   endpoints: (build) => ({
-    getTutorialsInRange: build.query<CalendarEntry[],
-    {startDatetime: number, endDatetime: number, course: string}>({
-      query(arg) {
-        const { startDatetime, endDatetime, course } = arg;
+    getTutorialsInRange: build.query<Tutorial[],
+    {startDatetime: number, endDatetime: number, courses: string}>({
+      query: (arg) => {
+        const { startDatetime, endDatetime, courses } = arg;
```
Comment (ericgmoon) [inline src/services/tutorials/index.ts:11]: I don't think `courses` as a comma-separated parameter is very intuitive. Shouldn't it be passed around as an array, which then gets transformed into a comma-separated input here?

**Review by ericgmoon — APPROVED** [review]
Approving with suggestions. Make sure you resolve merge conflict as well.

**Review by ericgmoon — APPROVED** [review]

**Pull request merged**